tests/Services.cpp

```diff
@@ -20,8 +20,10 @@
 using namespace opcua;
 using namespace std::literals::chrono_literals;
 
-template <typename F>
-constexpr auto syncWrapper(Client& client, F&& asyncClientFunction) {
+template <typename... Args>
+constexpr auto syncWrapper(
+    Client& client, std::future<std::vector<Variant>> (&asyncClientFunction)(Args...)
```

Comment:
Why do you favor this?

Reply:
This should also work for other async functions in the `services` namespace. We shouldn't hardcode this to the `callAsync` signature.

Reply:
I had to be more precise for the function signature to support both callAsync signatures. Though although the template parameter of the original syncWrapper looks generic, the implementation does expect the function to return a future.

We could change the signature of the callAsync function with the callback variant to something like `callAsyncWithCallback/callAsyncCallback`. This would make the distinction to the `call/callAsync` pair of the services more clear, then also `syncWrapper` for this variant could be renamed to something like `syncWrapperWithCallback/syncWrapperCallback`.

Do you prefer this solution? If yes, what is your preferred function name?

Reply:
`callAsyncWithCallback/callAsyncCallback` sounds good! 👍
This will be a temporary solution. Later, when we have a generic async model, we don't have to test each async model separately. Instead, we should have tests for the generic async handling and one simple test for the actual implementation like `callAsync`.

Reply:
Okay, I have changed it to callAsyncWithCallback for now :-)